internal/feedback/feedback.go

```diff
@@ -82,6 +82,127 @@ func (cf ConstraintFeedback) LogFeedback(logger *log.Logger) {
 	}
 }
 
+type brokenImport interface {
+	String() string
+}
+
+type modifiedImport struct {
+	source, branch, revision, version *gps.StringDiff
+	projectPath                       string
+}
+
+func (mi modifiedImport) String() string {
+	var pv string
+	var pr string
+	pp := mi.projectPath
+
+	var cr string
+	var cv string
+	cp := ""
+
+	if mi.revision != nil {
+		pr = fmt.Sprintf("(%s)", trimSHA(mi.revision.Previous))
+		cr = fmt.Sprintf("(%s)", trimSHA(mi.revision.Current))
+	}
+
+	if mi.version != nil {
+		pv = mi.version.Previous
+		cv = mi.version.Current
+	} else if mi.branch != nil {
+		pv = mi.branch.Previous
+		cv = mi.branch.Current
+	}
+
+	if mi.source != nil {
+		pp = fmt.Sprintf("%s(%s)", mi.projectPath, mi.source.Previous)
+		cp = fmt.Sprintf(" for %s(%s)", mi.projectPath, mi.source.Current)
+	}
+
+	// Warning: Unable to preserve imported lock VERSION/BRANCH (REV) for PROJECT(SOURCE). Locking in VERSION/BRANCH (REV) for PROJECT(SOURCE)
+	return fmt.Sprintf("%v %s for %s. Locking in %v %s%s",
+		pv,
+		pr,
+		pp,
+		cv,
+		cr,
+		cp,
+	)
+}
+
+type removedImport struct {
+	source, branch, revision, version *gps.StringDiff
+	projectPath                       string
+}
+
+func (ri removedImport) String() string {
+	var pr string
+	var pv string
+	pp := ri.projectPath
+
+	if ri.revision != nil {
+		pr = fmt.Sprintf("(%s)", trimSHA(ri.revision.Previous))
+	}
+
+	if ri.version != nil {
+		pv = ri.version.Previous
+	} else if ri.branch != nil {
+		pv = ri.branch.Previous
+	}
+
+	if ri.source != nil {
+		pp = fmt.Sprintf("%s(%s)", ri.projectPath, ri.source.Previous)
+	}
+
+	// Warning: Unable to preserve imported lock VERSION/BRANCH (REV) for PROJECT(SOURCE). Locking in VERSION/BRANCH (REV) for PROJECT(SOURCE)
+	return fmt.Sprintf("%v %s for %s. The project was removed from the lock because it is not used.",
+		pv,
+		pr,
+		pp,
+	)
+}
+
+// BrokenImportFeedback holds problematic lock feedback data
+type BrokenImportFeedback struct {
```

Comment:
Naming again: i think that neither "broken" nor "import" are the right terms here. i believe something like `ChangedLockFeedback` might be more appropriate.